File: 73-set-matrix-zeroes/73-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        vector<int> rows(matrix.size()), cols(matrix[0].size());
for( int i = 0; i < matrix.size(); i++){                 
     for(int j = 0; j< matrix[0].size(); j++){
	      if(matrix[i][j] == 0){
		      rows[i] = cols[j] = 1;       //updating hash tables
			 }
	   }
}
//Now updating the matrix in 2nd traversal
for( int i = 0; i < matrix.size(); i++){                 
     for(int j = 0; j< matrix[0].size(); j++){
	      if(rows[i] == 1 || cols[j] == 1){
		      matrix[i][j] = 0;       
			 }
	   }
}
    }
};

File: 73-set-matrix-zeroes/73-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        // ArrayList<Integer> row = new ArrayList<>();
        // ArrayList<Integer> col = new ArrayList<>();
    
        int rows = matrix.length;
        int cols = matrix[0].length;
        
        int [] row = new int[rows];
        int [] col = new int[cols];
        
        for(int i=0; i<rows; i++){
            for(int j=0; j<cols; j++){
                if (matrix[i][j]==0){
                    row[i]=1;
                    col[j]=1;
                    // row.set(i,0);
                    // col.set(j,0);
                }
            }
        }
        
        for(int i=0; i<rows; i++){
            for(int j=0; j<cols; j++){
              if(row[i]==1 || col[j]==1){
                  matrix[i][j] =0;
              }
            }
        }
    }
}